common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java

```diff
@@ -300,11 +306,70 @@ public static List<QuestInfo> getQuests() {
         return quests;
     }
 
+    public static List<QuestInfo> getQuestsSorted(QuestSortOrder sortOrder) {
+        return switch (sortOrder) {
+            case NONE -> quests;
+            case LEVEL -> quests.stream()
+                    .sorted(Comparator.comparing(questInfo -> questInfo.getLevel()))
+                    .toList();
+            case DISTANCE -> quests.stream().sorted(new LocationComparator()).toList();
+            case ALPHABETIC -> quests.stream()
+                    .sorted(Comparator.comparing(questInfo -> questInfo.getName()))
+                    .toList();
+        };
+    }
+
     public static List<List<String>> getDialogueHistory() {
         return dialogueHistory;
     }
 
     public static List<QuestInfo> getMiniQuests() {
         return miniQuests;
     }
+
+    public static Optional<Location> getLocationFromDescription(String description) {
+        Matcher matcher = COORDINATE_PATTERN.matcher(ComponentUtils.stripFormatting(description));
+        if (!matcher.matches()) return Optional.empty();
+
+        return Optional.of(new Location(
+                Integer.parseInt(matcher.group(1)),
+                Integer.parseInt(matcher.group(2)),
+                Integer.parseInt(matcher.group(3))));
+    }
+
+    private static class LocationComparator implements Comparator<QuestInfo> {
+        private static final Vec3 PLAYER_LOCATION = McUtils.player().position();
+
+        private double getDistance(Optional<Location> loc) {
```

Comment:
Is Qodana's suggestion relevant?

Reply:
Ah, found it:

>  'Optional' used as field or parameter type: common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java#L345
`Optional` used as type for parameter 'loc'

Reply:
(But not in the code review GUI, had to search in the GHA logs...)

No, I don't think that's relevant. I mean, it's kind of the point that we should take in an Optional and return 0 if it is empty. I get it why one should avoid Optional in fields, but I have a hard time understanding why they should not be passed as arguments. Seems quite valid to me.